app/models/user.rb

```diff
@@ -29,6 +29,8 @@
 class User < ActiveRecord::Base
   include DeviseTokenAuth::Concerns::User
 
+  after_update :handle_subscription
+
```

Comment:
I wanted to do a `before_update` here so I can handle subscriptions with simpler dirty methods 

```
  def handle_subscription
    if email_changed? || email_subscription_changed? && email_subcription_was
      unsubscribe_previous_email_from_newsletter 
    end

    subscribe_to_newsletter if email_subscription_change
  end
```